napari/layers/points/points.py

```diff
@@ -412,6 +793,162 @@ def current_face_color(self, face_color: ColorType) -> None:
         self.events.face_color()
         self.events.highlight()
 
+    @property
+    def face_color_mode(self):
+        """str: Face color setting mode
+
+        DIRECT (default mode) allows each point to be set arbitrarily
+
+        CYCLE allows the color to be set via a color cycle over an attribute
+
+        COLORMAP allows color to be set via a color map over an attribute
+        """
+        return str(self._face_color_mode)
+
+    @face_color_mode.setter
+    def face_color_mode(self, face_color_mode):
+        if isinstance(face_color_mode, str):
+            face_color_mode = ColorMode(face_color_mode)
+
+        if face_color_mode == ColorMode.DIRECT:
+            self._face_color_mode = face_color_mode
+        elif face_color_mode in (ColorMode.CYCLE, ColorMode.COLORMAP):
+            if self._face_color_property == '':
+                if self.properties:
+                    self._face_color_property = next(iter(self.properties))
+                    warnings.warn(
+                        'Face color was not set, setting to: %s'
+                        % self._face_color_property
+                    )
+                else:
+                    raise ValueError(
+                        'There must be a valid Points.properties to use %s'
+                        % face_color_mode
+                    )
+
+            # ColorMode.COLORMAP can only be applied to numeric properties
+            if (face_color_mode == ColorMode.COLORMAP) and not issubclass(
+                self.properties[self._face_color_property].dtype.type,
+                np.number,
+            ):
+                raise TypeError(
+                    'selected property must be numeric to use ColorMode.COLORMAP'
+                )
+            self._face_color_mode = face_color_mode
+            self.refresh_colors()
+
+    def refresh_colors(self, update_color_mapping: bool = True):
+        """ Calculate and update face and edge colors if using a cycle or color map
+
+            Parameters
+            ----------
+            update_color_mapping : bool
+                If set to True, the function will recalculate the
+                color cycle map or colormap (whichever is being used).
+                If set to False, the function will use the current
+                color cycle map or color map. For example, if you
+                are adding/modifying points and want them to be
+                colored with the same mapping as the other
+                points (i.e., the new points shouldn't affect
+                the color cycle map or colormap), set update_color_mapping=False.
+                Default value is True.
+        """
```

Comment:
Incorrect docstring formatting again... :grimacing:

Reply:
Dang it! fixed.